### PR TITLE
Home task 3

### PR DESCRIPTION
**Pull request by NureBerehovyiDaniil**
Try hard

### CODE REVIEW

File: BLL/obj/BLL.csproj.nuget.dgspec.json
Comment (vemontd1) [inline BLL/obj/BLL.csproj.nuget.dgspec.json:1]: Fix .gitignore to exclude "**/obj/**" and "**/bin/**"

File: BLL/Services/BookingService.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        public async Task<List<Booking>> GetBookingsByMember(Guid memberId)
+        {
+            return await _memberService.BookingRepository.GetBookingsByMember(memberId);
+        }
+
+        public async Task<List<Booking>> GetBookingsByClass(Guid classId)
+        {
+            return await _unitOfWork.BookingRepository.GetBookingsByClass(classId);
```
Comment (vemontd1) [inline BLL/Services/BookingService.cs:34]: Where _unitOfWork comes from?

File: BLL/Services/MemberService.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        public async Task<bool> CheckMemberAttendance(Guid memberId, DateTime date)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task RecordMemberAttendance(Guid memberId, DateTime date)
+        {
+            throw new NotImplementedException();
+        }
```
Comment (vemontd1) [inline BLL/Services/MemberService.cs:57]: Why not implemented?

File: UI/ConsoleManagers/SubscriptionConsoleManager.cs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+                else
+                {
+                    Console.WriteLine("Subscription not found.");
+                }
+            }
+            else
+            {
+                Console.WriteLine("Invalid subscription ID format.");
+            }
+        }
```
Comment (vemontd1) [inline UI/ConsoleManagers/SubscriptionConsoleManager.cs:223]: Methods shouldn't be larger than 40 lines. Better to refactor this, move some common logic to private methods

File: UI/ConsoleManagers/TrainerConsoleManager.cs [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+            foreach (var trainer in allTrainers)
+            {
+                if (trainer.AvailableDates.Contains(date))
+                {
+                    availableTrainers.Add(trainer);
+                }
+            }
+
+            return availableTrainers;
```
Comment (vemontd1) [inline UI/ConsoleManagers/TrainerConsoleManager.cs:100]: I think this can be simplified by using GetByPredicate

**Review by vemontd1 — CHANGES_REQUESTED** [review]

File: BLL/obj/BLL.csproj.nuget.dgspec.json
Comment (PetrPoling) [reply BLL/obj/BLL.csproj.nuget.dgspec.json:1]: Thank you for check, already fixed

File: BLL/Services/MemberService.cs (hunk 2 above)
Comment (PetrPoling) [reply BLL/Services/MemberService.cs]: Tried to figure out how does member service works, Already implemented.

File: UI/ConsoleManagers/TrainerConsoleManager.cs (hunk 4 above)
Comment (PetrPoling) [reply UI/ConsoleManagers/TrainerConsoleManager.cs:100]: Checked and simplified.

File: BLL/obj/BLL.csproj.nuget.dgspec.json
Comment (PetrPoling) [reply BLL/obj/BLL.csproj.nuget.dgspec.json:1]: Already done for all variants of obj and bin data

File: BLL/Services/BookingService.cs (hunk 1 above)
Comment (PetrPoling) [reply BLL/Services/BookingService.cs]: Forget to change the name of service, Already debugged and fixed

**Review by vemontd1 — APPROVED** [review]

**Pull request merged**